quantization/src/encoded_vectors.rs

```diff
@@ -37,6 +38,7 @@ impl DistanceType {
     pub fn distance(&self, a: &[f32], b: &[f32]) -> f32 {
         match self {
             DistanceType::Dot => a.iter().zip(b.iter()).map(|(a, b)| a * b).sum(),
+            DistanceType::L1 => a.iter().zip(b.iter()).map(|(a, b)| (a - b).abs()).sum(),
             DistanceType::L2 => a.iter().zip(b.iter()).map(|(a, b)| (a - b) * (a - b)).sum(),
```

Comment:
Though fine, `.iter()` is redundant here:

```suggestion
            DistanceType::Dot => a.iter().zip(b).map(|(a, b)| a * b).sum(),
            DistanceType::L1 => a.iter().zip(b).map(|(a, b)| (a - b).abs()).sum(),
            DistanceType::L2 => a.iter().zip(b).map(|(a, b)| (a - b) * (a - b)).sum(),
```